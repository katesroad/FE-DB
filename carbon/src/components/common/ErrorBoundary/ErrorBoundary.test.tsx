import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { ErrorBoundaryProvider } from '.'

const errorMsg = 'We have a bug!!!'
const EleMayHaveError = () => {
  throw new Error(errorMsg)
}
const renderUI = (ui: React.ReactElement, pathname: string) => {
  return render(
    <MemoryRouter initialEntries={[{ pathname }]}>
      <ErrorBoundaryProvider>{ui}</ErrorBoundaryProvider>
    </MemoryRouter>
  )
}

test('use <ErrorBoundary /> at home page', () => {
  const { container } = renderUI(<EleMayHaveError />, '/')
  expect(
    screen.queryByRole('link', { name: /to go home/i })
  ).not.toBeInTheDocument()
  expect(container).toMatchSnapshot()
})

test('use <ErrorBoundary /> at other pages', () => {
  const { container } = renderUI(<EleMayHaveError />, '/regions')
  expect(
    screen.queryByRole('link', { name: /to go home/i })
  ).toBeInTheDocument()
  expect(container).toMatchSnapshot()
})
