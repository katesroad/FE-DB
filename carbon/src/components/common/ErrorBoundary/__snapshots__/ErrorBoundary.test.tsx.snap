// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`use <ErrorBoundary /> at home page 1`] = `
<div>
  <div
    class="error-boundary"
  >
    <div
      class="ui-error"
      role="alert"
    >
      <h2>
        Something went wrong:
      </h2>
      <pre>
        We have a bug!!!
      </pre>
      <p
        class="error-boundary__solution"
      >
        <button
          class="ui-button"
        >
          refresh
        </button>
      </p>
    </div>
  </div>
</div>
`;

exports[`use <ErrorBoundary /> at other pages 1`] = `
<div>
  <div
    class="error-boundary"
  >
    <div
      class="ui-error"
      role="alert"
    >
      <h2>
        Something went wrong:
      </h2>
      <pre>
        We have a bug!!!
      </pre>
      <p
        class="error-boundary__solution"
      >
        <a
          class="ui-go-home"
          href="/"
        >
          <span>
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 19l-7-7m0 0l7-7m-7 7h18"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            <span>
              Go To Home
            </span>
          </span>
        </a>
        <button
          class="ui-button"
        >
          refresh
        </button>
      </p>
    </div>
  </div>
</div>
`;
